### PR TITLE
docs: addition of multi language support into readme

### LINKED ISSUE #5502: docs: APISIX supports multi-language architecture diagram
### Improved or typo fixed.

Not yet. Added to the documentation, such as chapters on multi-language and WASM.

### Solution

image

### PR DESCRIPTION
**Pull request by bisakhmondal**
### What this PR does / why we need it:

closes #5502 
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by bisakhmondal** [comment]
cc @membphis

**Review by membphis — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -146,6 +146,12 @@ A/B testing, canary release, blue-green deployment, limit rate, defense against
   - Custom load balancing algorithms: You can use custom load balancing algorithms during the `balancer` phase.
   - Custom routing: Support users to implement routing algorithms themselves.
 
+- **Multi-Language support**
+  - Apache APISIX is a multi-language gateway for plugin development and provides support via `WASM` and `RPC`.
+  ![Multi Language Support into Apache APISIX](docs/assets/images/apisix-multi-lang-support.png)
+  - The WASM or WebAssembly, is the modern way. If we look at the left part, APISIX can load and run WASM bytecode via APISIX wasm plugin, so developers only need to write the code according to the SDK and then compile it into a WASM bytecode file and put it in the APISIX folder. As of now, APISIX has support for Golang SDK and Rust SDK - more language SDKs will be supported in the future.
```
Comment (membphis) [inline README.md:152]: we can add a link for `Golang SDK` and `Rust SDK` later.

File: README.md [hunk 2]
```diff
@@ -146,6 +146,12 @@ A/B testing, canary release, blue-green deployment, limit rate, defense against
...
+  - The RPC way, is a traditional way. If we look at the right part, developers can choose the language according to their needs. After starting an independent process with the RPC, it exchanges data with APISIX through local RPC communication. Till this moment, APISIX has support for Java， Golang, Python and Node.js.
```
Comment (membphis) [inline README.md:153]: same issue, we can add a link to `Golang`, `Python` and `Java`.

eg: the Java should be linked to https://github.com/apache/apisix-java-plugin-runner

File: README.md (hunk 1 above)
Comment (bisakhmondal) [reply README.md]: Hi @membphis, one question - if the wasm plugin of apache apisix just runs the wasm bytecode why are we limited only to the Go and Rust? At the end of the day, it's the bytecode instructions that we need to run on wasm, right?

https://webassembly.org/getting-started/developers-guide/

This official URL shows the list of languages where the programs can e compiled for wasm virtual machine. Thanks

**Review by tokers — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -146,6 +146,12 @@ A/B testing, canary release, blue-green deployment, limit rate, defense against
   - Custom load balancing algorithms: You can use custom load balancing algorithms during the `balancer` phase.
   - Custom routing: Support users to implement routing algorithms themselves.
 
+- **Multi-Language support**
+  - Apache APISIX is a multi-language gateway for plugin development and provides support via `WASM` and `RPC`.
+  ![Multi Language Support into Apache APISIX](docs/assets/images/apisix-multi-lang-support.png)
+  - The WASM or WebAssembly, is the modern way. If we look at the left part, APISIX can load and run WASM bytecode via APISIX wasm plugin, so developers only need to write the code according to the SDK and then compile it into a WASM bytecode file and put it in the APISIX folder. As of now, APISIX has support for [Golang SDK](https://github.com/golang/go/wiki/WebAssembly#getting-started) and [Rust SDK](https://developer.mozilla.org/en-US/docs/WebAssembly/Rust_to_wasm) - more language SDKs will be supported in the future.
```
Comment (spacewander) [inline README.md:152]: The SDK should be https://github.com/proxy-wasm/spec#sdks.
And there are not just Go/Rust.
And there is no requirement to `put it in the APISIX folder`.
See https://github.com/apache/apisix/blob/master/docs/en/latest/wasm.md

**Review by tzssangglass — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (membphis) [reply README.md:152]: ping @bisakhmondal

**Review by moonming — APPROVED** [review]

**Comment by juzhiyuan** [comment]
@bisakhmondal Hi, please resolve those conflicts, thanks!

File: README.md (hunk 3 above)
Comment (bisakhmondal) [reply README.md]: Ah, I got it now. The sdk means - the proxy wasm SDK. Updated accordingly. Is it okay now?
cc @spacewander

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**